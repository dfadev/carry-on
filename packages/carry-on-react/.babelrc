{
  "presets": [
    ["env", {
      "modules": false,
      "targets": { "browsers": ["last 2 versions"] }
    }],
    "react",
    "stage-2",
    "optimized"
  ],
  "env": {
    "development": {
      "plugins": ["external-helpers"],
    },
    "production": {
      "plugins": ["external-helpers"],
    },
    "test": {
      "presets": ["env", "react", "stage-2"],
      "plugins": ["transform-es2015-modules-commonjs"]
    }
  }
}
