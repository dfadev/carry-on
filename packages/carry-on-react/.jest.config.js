module.exports = {
  setupTestFrameworkScriptFile: "./test/jest.setup.js",
  verbose: true,
  testPathIgnorePatterns: [
    "<rootDir>/build/",
    "<rootDir>/node_modules/",
    "<rootDir>/examples/",
    "<rootDir>/lib",
    "<rootDir>/src/jest.setup.js"
  ],
  collectCoverage: true,
  coverageDirectory: "<rootDir>/coverage/",
  moduleNameMapper: {
    "^carry-on-utils$":
      "<rootDir>/node_modules/carry-on-utils/dist/carry-on-utils.esm.js",
    "^carry-on-store$":
      "<rootDir>/node_modules/carry-on-store/dist/carry-on-store.esm.js",
  }
};
