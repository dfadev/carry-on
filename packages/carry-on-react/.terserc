module.exports = {
  ecma: "5",
  warnings: "verbose",
  compress: {
    hoist_funs: "true",
    drop_console: "true",
    passes: "2",
    unsafe: "true",
    unsafe_arrows: "true",
    unsafe_comps: "true",
    unsafe_methods: "true",
    unsafe_proto: "true",
    warnings: "true",
    pure_getters: "true"
  },
  mangle: {
    module: "true",
    properties: {
      regex: /^_/
    }
  },
  module: "true",
  output: {
    semicolons: "false"
  }
};
