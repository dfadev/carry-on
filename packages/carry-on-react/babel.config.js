module.exports = {
  presets: [
    [
      "@babel/preset-env",
      {
        modules: false,
        targets: {
          browsers: ["last 2 versions"]
        }
      }
    ],
    "@babel/preset-react"
  ],
  env: {
    test: {
      presets: ["@babel/preset-env", "@babel/preset-react"],
      plugins: ["@babel/plugin-transform-modules-commonjs"]
    }
  },
  plugins: ["@babel/plugin-proposal-class-properties"]
};
