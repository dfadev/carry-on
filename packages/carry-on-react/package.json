{
  "name": "carry-on-react",
  "version": "0.0.13",
  "description": "A Store for React",
  "source": "./src/index.js",
  "module": "./es/carry-on-react.js",
  "scripts": {
    "clean": "rimraf lib dist es coverage",
    "format": "prettier --write \"{src,test}/**/*.js\"",
    "format:check": "prettier --list-different \"{src,test}/**/*.js\"",
    "lint": "eslint src test",
    "pretest": "npm run build",
    "test": "cross-env BABEL_ENV=test jest --config ./.jest.config.js",
    "test:watch": "npm test -- --watchAll",
    "test:cov": "npm test -- --coverage",
    "build": "rollup -c rollup.config.js",
    "build-dev": "rollup -c rollup.config.js --configDebug",
    "watch": "npm run build -- -w"
  },
  "author": "Russ Panula <russ@dfa.net>",
  "license": "MIT",
  "peerDependencies": {
    "react": "^16.4.2",
    "react-dom": "^16.4.2"
  },
  "devDependencies": {
    "@babel/core": "^7.0.0",
    "@babel/plugin-proposal-class-properties": "^7.2.3",
    "@babel/plugin-transform-modules-commonjs": "^7.2.0",
    "@babel/preset-env": "^7.2.3",
    "@babel/preset-react": "^7.0.0",
    "babel-core": "^7.0.0-bridge.0",
    "babel-eslint": "^9.0.0",
    "babel-jest": "^23.4.2",
    "cross-env": "^5.2.0",
    "eslint": "^5.10.0",
    "eslint-config-airbnb": "^17.1.0",
    "eslint-config-prettier": "^3.0.1",
    "eslint-plugin-import": "^2.14.0",
    "eslint-plugin-jsx-a11y": "^6.1.2",
    "eslint-plugin-react": "^7.11.1",
    "jest": "^23.5.0",
    "prettier": "^1.14.2",
    "react": "^16.7.0",
    "react-dom": "^16.7.0",
    "react-testing-library": "^5.4.2",
    "rimraf": "^2.6.2",
    "rollup": "^0.64.1",
    "rollup-plugin-babel": "^4.0.1",
    "rollup-plugin-eslint": "^5.0.0",
    "snapshot-diff": "^0.4.1"
  },
  "dependencies": {
    "carry-on-store": "0.0.13",
    "carry-on-utils": "0.0.13",
    "hoist-non-react-statics": "^3.2.1"
  },
  "gitHead": "e21942a1d6e23d964f9b74bd1c9c3744cb7b3c18"
}
