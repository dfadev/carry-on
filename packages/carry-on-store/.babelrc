{
  "presets": [
    ["env", {
      "modules": false,
      "targets": { "browsers": ["last 2 versions"] }
    }],
    "stage-2",
    "optimized"
  ],
  "env": {
    "development": {
      "plugins": ["external-helpers"],
    },
    "production": {
      "plugins": ["external-helpers"],
    },
    "test": {
      "presets": ["env", "stage-2"]
    }
  }
}
