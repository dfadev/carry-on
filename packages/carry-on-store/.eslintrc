{
  "parser": "babel-eslint",
  "extends": ["airbnb-base", "prettier"],
  "env": { "es6": true, "browser": true },
  "rules": {
    "no-empty-pattern": "off",
    "no-debugger": "warn",
    "no-unused-expressions": "off",
    "no-unused-vars": "warn",
    "no-plusplus": "off",
    "import/no-unresolved": "off",
    "no-restricted-syntax": "off",
    "no-underscore-dangle": "off",
    "prefer-template": "off",
    "spaced-comment": "off",
    "no-param-reassign": "off",
    "no-return-assign": "off",
    "no-multi-assign": "off",
    "one-var": "off",
    "prefer-destructuring": "off",
  }
}
