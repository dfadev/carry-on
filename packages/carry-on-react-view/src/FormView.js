/* eslint-disable react/jsx-props-no-spreading */
import React from "react";
import { Form, FormState } from "carry-on-react-forms";
import { withState, withNodesToProps } from "carry-on-react";
import { deproxify } from "carry-on-utils";
import SectionView from "./SectionView";

const components = {
  select: ({
    components: { FormViewBox = "div", SectionBox = "div" } = {}
  }) => ({
    FormViewBox,
    SectionBox
  })
};

const FormView = ({
  from,
  store = from,
  id = "form",
  FormViewBox: RootFormViewBox,
  SectionBox: RootSectionBox,
  register = [],
  children,
  onMount,
  onUnmount,
  onValidate,
  onSubmit,
  onReset,
  ...rest
}) => (
  <Form
    id={id}
    register={[
      ...register,
      {
        state: {
          [id]: { ...rest }
        }
      }
    ]}
    store={store}
    onMount={onMount}
    onUnmount={onUnmount}
    onValidate={onValidate}
    onSubmit={onSubmit}
    onReset={onReset}
  >
    <FormState form={id}>
      {({
        view,
        section,
        sections = [],
        components: {
          FormViewBox = RootFormViewBox,
          SectionBox = RootSectionBox
        } = {}
      }) =>
        sections && sections.length > 0 ? (
          <FormViewBox {...view}>
            {sections.map((sectionEntry, j) => (
              <SectionBox
                key={j /* eslint-disable-line react/no-array-index-key */}
                {...deproxify(section)}
                {...deproxify(sectionEntry)}
              >
                <SectionView {...deproxify(sectionEntry)} />
              </SectionBox>
            ))}
          </FormViewBox>
        ) : null
      }
    </FormState>
    {children}
  </Form>
);

FormView.composes = ["initialValues", "sections", "fields", "view", "children"];

export default withState(components)(withNodesToProps(FormView));
