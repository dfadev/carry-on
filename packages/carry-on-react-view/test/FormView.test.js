/**
 * @jest-environment jsdom
 */
import React from "react";
import { register, initStores } from "carry-on-store";
import { Store, State, Register } from "carry-on-react";
import { render } from "@testing-library/react";
import FormView from "../src/FormView";
import {
  InitialValues,
  Sections,
  Section,
  Fields,
  Field
} from "../src/Composables";

describe("FormView", () => {
  beforeEach(() => initStores());

  it("<FormView /> renders", () => {
    const { asFragment } = render(<FormView />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders example 1", () => {
    const view = {
      id: "formView1",
      initialValues: {
        abc: "123",
        def: "345",
        ghi: undefined,
        jkl: null
      },
      sections: [
        {
          title: "Section 1",
          layout: ["abc", "def"]
        },
        {
          title: "Section 2",
          layout: [["ghi"], ["jkl"]]
        }
      ],
      fields: {
        abc: {
          label: "ABC",
          tooltip: () => <div>ABC tooltip</div>
        }
      },
      components: {
        FormViewBox: ({ children }) => (
          <div style={{ border: "1px solid purple", padding: "16px" }}>
            <h2>FormViewBox</h2>
            {children}
          </div>
        ),
        SectionBox: ({ section: { title }, children }) => (
          <div
            style={{
              border: "1px solid blue",
              margin: "16px",
              padding: "16px"
            }}
          >
            <h2>SectionBox: {title}</h2>
            {children}
          </div>
        ),
        View: ({ children }) => (
          <div
            style={{ border: "1px solid red", margin: "16px", padding: "16px" }}
          >
            <div>
              <h2>View</h2>
            </div>
            <div style={{ display: "flex" }}>{children}</div>
          </div>
        ),
        ViewItem: ({ children }) => (
          <div
            style={{
              border: "1px solid white",
              margin: "16px",
              padding: "16px"
            }}
          >
            <div>
              <h2>ViewItem</h2>
            </div>
            <div style={{ display: "block" }} margin="16px" padding="16px">
              {children}
            </div>
          </div>
        ),
        editors: {}
      }
    };

    const { asFragment } = render(
      <Store id="formViewStore">
        <FormView {...view} />
      </Store>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders example 2", () => {
    register("formViewStore2", {
      state: {
        components: {
          FormViewBox: ({ children }) => (
            <div style={{ border: "1px solid purple", padding: "16px" }}>
              <h2>FormViewBox</h2>
              {children}
            </div>
          ),
          SectionBox: ({ children }) => (
            <div
              style={{
                border: "1px solid blue",
                margin: "16px",
                padding: "16px"
              }}
            >
              <h2>SectionBox</h2>
              {children}
            </div>
          ),
          View: ({ children }) => (
            <div
              style={{
                border: "1px solid red",
                margin: "16px",
                padding: "16px"
              }}
            >
              <div>
                <h2>View</h2>
              </div>
              <div style={{ display: "flex" }}>{children}</div>
            </div>
          ),
          ViewItem: ({ children }) => (
            <div
              style={{
                border: "1px solid white",
                margin: "16px",
                padding: "16px"
              }}
            >
              <div>
                <h2>ViewItem</h2>
              </div>
              <div style={{ display: "block" }} margin="16px" padding="16px">
                {children}
              </div>
            </div>
          ),
          editors: {}
        }
      }
    });

    const view = {
      id: "formView2",
      initialValues: {
        abc: "123",
        def: "345",
        ghi: undefined,
        jkl: null
      },
      sections: [
        {
          title: "Section 1",
          layout: [["abc", "def"]]
        },
        {
          title: "Section 2",
          layout: [["ghi"], ["jkl"]]
        }
      ],
      fields: {
        abc: {
          label: "ABC"
        }
      }
    };

    const { asFragment } = render(
      <Store id="formViewStore2">
        <FormView {...view} />
      </Store>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders example 3", () => {
    const components = {
      components: {
        FormViewBox: ({ children }) => (
          <div style={{ border: "1px solid purple", padding: "16px" }}>
            <h2>FormViewBox</h2>
            {children}
          </div>
        ),
        SectionBox: ({ section: { title }, children }) => (
          <div
            style={{
              border: "1px solid blue",
              margin: "16px",
              padding: "16px"
            }}
          >
            <h2>SectionBox: {title}</h2>
            {children}
          </div>
        ),
        View: ({ children }) => (
          <div
            style={{ border: "1px solid red", margin: "16px", padding: "16px" }}
          >
            <div>
              <h2>View</h2>
            </div>
            <div style={{ display: "flex" }}>{children}</div>
          </div>
        ),
        ViewItem: ({ children }) => (
          <div
            style={{
              border: "1px solid white",
              margin: "16px",
              padding: "16px"
            }}
          >
            <div>
              <h2>ViewItem</h2>
            </div>
            <div style={{ display: "block" }} margin="16px" padding="16px">
              {children}
            </div>
          </div>
        ),
        editors: {}
      }
    };

    const { asFragment } = render(
      <Store id="formViewStoreComposing">
        <State>
          <Register>{components}</Register>
        </State>
        <FormView id="formViewComposing">
          <InitialValues>
            {{
              abc: "abc"
            }}
          </InitialValues>
          <Sections>
            <Section title="Section 1">{["abc", "def"]}</Section>
            <Section title="Section 2">{[["ghi"], ["jkl"]]}</Section>
          </Sections>
          <Fields>
            <Field name="abc">
              {{
                label: "ABC",
                tooltip: "The ABC Tooltip"
              }}
            </Field>
          </Fields>
        </FormView>
      </Store>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
