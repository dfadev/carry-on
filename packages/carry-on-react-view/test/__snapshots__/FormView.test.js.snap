// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormView <FormView /> renders 1`] = `
<DocumentFragment>
  <form
    id="form"
  >
    <div />
  </form>
</DocumentFragment>
`;

exports[`FormView renders example 1 1`] = `
<DocumentFragment>
  <form
    id="formViewStore.formView1"
  >
    <div
      style="border: 1px solid purple; padding: 16px;"
    >
      <h2>
        FormViewBox
      </h2>
      <div
        style="border: 1px solid blue; margin: 16px; padding: 16px;"
      >
        <h2>
          SectionBox: Section 1
        </h2>
        <div
          style="border: 1px solid red; margin: 16px; padding: 16px;"
        >
          <div>
            <h2>
              View
            </h2>
          </div>
          <div
            style="display: flex;"
          >
            <div
              style="border: 1px solid white; margin: 16px; padding: 16px;"
            >
              <div>
                <h2>
                  ViewItem
                </h2>
              </div>
              <div
                margin="16px"
                padding="16px"
                style="display: block;"
              >
                <div>
                  <div>
                    <label
                      for="formViewStore.formView1.abc"
                    >
                      ABC
                    </label>
                  </div>
                  <div>
                    <input
                      id="formViewStore.formView1.abc"
                      name="formViewStore.formView1.abc"
                      value="123"
                    />
                  </div>
                  <div>
                    ABC tooltip
                  </div>
                  <br />
                </div>
              </div>
            </div>
            <div
              style="border: 1px solid white; margin: 16px; padding: 16px;"
            >
              <div>
                <h2>
                  ViewItem
                </h2>
              </div>
              <div
                margin="16px"
                padding="16px"
                style="display: block;"
              >
                <div>
                  <div>
                    <label
                      for="formViewStore.formView1.def"
                    >
                      def
                    </label>
                  </div>
                  <div>
                    <input
                      id="formViewStore.formView1.def"
                      name="formViewStore.formView1.def"
                      value="345"
                    />
                  </div>
                  <br />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        style="border: 1px solid blue; margin: 16px; padding: 16px;"
      >
        <h2>
          SectionBox: Section 2
        </h2>
        <div
          style="border: 1px solid red; margin: 16px; padding: 16px;"
        >
          <div>
            <h2>
              View
            </h2>
          </div>
          <div
            style="display: flex;"
          >
            <div
              style="border: 1px solid white; margin: 16px; padding: 16px;"
            >
              <div>
                <h2>
                  ViewItem
                </h2>
              </div>
              <div
                margin="16px"
                padding="16px"
                style="display: block;"
              >
                <div>
                  <div>
                    <label
                      for="formViewStore.formView1.ghi"
                    >
                      ghi
                    </label>
                  </div>
                  <div>
                    <input
                      id="formViewStore.formView1.ghi"
                      name="formViewStore.formView1.ghi"
                      value=""
                    />
                  </div>
                  <br />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          style="border: 1px solid red; margin: 16px; padding: 16px;"
        >
          <div>
            <h2>
              View
            </h2>
          </div>
          <div
            style="display: flex;"
          >
            <div
              style="border: 1px solid white; margin: 16px; padding: 16px;"
            >
              <div>
                <h2>
                  ViewItem
                </h2>
              </div>
              <div
                margin="16px"
                padding="16px"
                style="display: block;"
              >
                <div>
                  <div>
                    <label
                      for="formViewStore.formView1.jkl"
                    >
                      jkl
                    </label>
                  </div>
                  <div>
                    <input
                      id="formViewStore.formView1.jkl"
                      name="formViewStore.formView1.jkl"
                      value=""
                    />
                  </div>
                  <br />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
</DocumentFragment>
`;

exports[`FormView renders example 2 1`] = `
<DocumentFragment>
  <form
    id="formViewStore2.formView2"
  >
    <div
      style="border: 1px solid purple; padding: 16px;"
    >
      <h2>
        FormViewBox
      </h2>
      <div
        style="border: 1px solid blue; margin: 16px; padding: 16px;"
      >
        <h2>
          SectionBox
        </h2>
        <div
          style="border: 1px solid red; margin: 16px; padding: 16px;"
        >
          <div>
            <h2>
              View
            </h2>
          </div>
          <div
            style="display: flex;"
          >
            <div
              style="border: 1px solid white; margin: 16px; padding: 16px;"
            >
              <div>
                <h2>
                  ViewItem
                </h2>
              </div>
              <div
                margin="16px"
                padding="16px"
                style="display: block;"
              >
                <div>
                  <div>
                    <label
                      for="formViewStore2.formView2.abc"
                    >
                      ABC
                    </label>
                  </div>
                  <div>
                    <input
                      id="formViewStore2.formView2.abc"
                      name="formViewStore2.formView2.abc"
                      value="123"
                    />
                  </div>
                  <br />
                </div>
              </div>
            </div>
            <div
              style="border: 1px solid white; margin: 16px; padding: 16px;"
            >
              <div>
                <h2>
                  ViewItem
                </h2>
              </div>
              <div
                margin="16px"
                padding="16px"
                style="display: block;"
              >
                <div>
                  <div>
                    <label
                      for="formViewStore2.formView2.def"
                    >
                      def
                    </label>
                  </div>
                  <div>
                    <input
                      id="formViewStore2.formView2.def"
                      name="formViewStore2.formView2.def"
                      value="345"
                    />
                  </div>
                  <br />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        style="border: 1px solid blue; margin: 16px; padding: 16px;"
      >
        <h2>
          SectionBox
        </h2>
        <div
          style="border: 1px solid red; margin: 16px; padding: 16px;"
        >
          <div>
            <h2>
              View
            </h2>
          </div>
          <div
            style="display: flex;"
          >
            <div
              style="border: 1px solid white; margin: 16px; padding: 16px;"
            >
              <div>
                <h2>
                  ViewItem
                </h2>
              </div>
              <div
                margin="16px"
                padding="16px"
                style="display: block;"
              >
                <div>
                  <div>
                    <label
                      for="formViewStore2.formView2.ghi"
                    >
                      ghi
                    </label>
                  </div>
                  <div>
                    <input
                      id="formViewStore2.formView2.ghi"
                      name="formViewStore2.formView2.ghi"
                      value=""
                    />
                  </div>
                  <br />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          style="border: 1px solid red; margin: 16px; padding: 16px;"
        >
          <div>
            <h2>
              View
            </h2>
          </div>
          <div
            style="display: flex;"
          >
            <div
              style="border: 1px solid white; margin: 16px; padding: 16px;"
            >
              <div>
                <h2>
                  ViewItem
                </h2>
              </div>
              <div
                margin="16px"
                padding="16px"
                style="display: block;"
              >
                <div>
                  <div>
                    <label
                      for="formViewStore2.formView2.jkl"
                    >
                      jkl
                    </label>
                  </div>
                  <div>
                    <input
                      id="formViewStore2.formView2.jkl"
                      name="formViewStore2.formView2.jkl"
                      value=""
                    />
                  </div>
                  <br />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
</DocumentFragment>
`;

exports[`FormView renders example 3 1`] = `
<DocumentFragment>
  <form
    id="formViewStoreComposing.formViewComposing"
  >
    <div
      style="border: 1px solid purple; padding: 16px;"
    >
      <h2>
        FormViewBox
      </h2>
      <div
        style="border: 1px solid blue; margin: 16px; padding: 16px;"
      >
        <h2>
          SectionBox: Section 1
        </h2>
        <div
          style="border: 1px solid red; margin: 16px; padding: 16px;"
        >
          <div>
            <h2>
              View
            </h2>
          </div>
          <div
            style="display: flex;"
          >
            <div
              style="border: 1px solid white; margin: 16px; padding: 16px;"
            >
              <div>
                <h2>
                  ViewItem
                </h2>
              </div>
              <div
                margin="16px"
                padding="16px"
                style="display: block;"
              >
                <div>
                  <div>
                    <label
                      for="formViewStoreComposing.formViewComposing.abc"
                    >
                      ABC
                    </label>
                  </div>
                  <div>
                    <input
                      id="formViewStoreComposing.formViewComposing.abc"
                      name="formViewStoreComposing.formViewComposing.abc"
                      value="abc"
                    />
                  </div>
                  The ABC Tooltip
                  <br />
                </div>
              </div>
            </div>
            <div
              style="border: 1px solid white; margin: 16px; padding: 16px;"
            >
              <div>
                <h2>
                  ViewItem
                </h2>
              </div>
              <div
                margin="16px"
                padding="16px"
                style="display: block;"
              >
                <div>
                  <div>
                    <label
                      for="formViewStoreComposing.formViewComposing.def"
                    >
                      def
                    </label>
                  </div>
                  <div>
                    <input
                      id="formViewStoreComposing.formViewComposing.def"
                      name="formViewStoreComposing.formViewComposing.def"
                      value=""
                    />
                  </div>
                  <br />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        style="border: 1px solid blue; margin: 16px; padding: 16px;"
      >
        <h2>
          SectionBox: Section 2
        </h2>
        <div
          style="border: 1px solid red; margin: 16px; padding: 16px;"
        >
          <div>
            <h2>
              View
            </h2>
          </div>
          <div
            style="display: flex;"
          >
            <div
              style="border: 1px solid white; margin: 16px; padding: 16px;"
            >
              <div>
                <h2>
                  ViewItem
                </h2>
              </div>
              <div
                margin="16px"
                padding="16px"
                style="display: block;"
              >
                <div>
                  <div>
                    <label
                      for="formViewStoreComposing.formViewComposing.ghi"
                    >
                      ghi
                    </label>
                  </div>
                  <div>
                    <input
                      id="formViewStoreComposing.formViewComposing.ghi"
                      name="formViewStoreComposing.formViewComposing.ghi"
                      value=""
                    />
                  </div>
                  <br />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          style="border: 1px solid red; margin: 16px; padding: 16px;"
        >
          <div>
            <h2>
              View
            </h2>
          </div>
          <div
            style="display: flex;"
          >
            <div
              style="border: 1px solid white; margin: 16px; padding: 16px;"
            >
              <div>
                <h2>
                  ViewItem
                </h2>
              </div>
              <div
                margin="16px"
                padding="16px"
                style="display: block;"
              >
                <div>
                  <div>
                    <label
                      for="formViewStoreComposing.formViewComposing.jkl"
                    >
                      jkl
                    </label>
                  </div>
                  <div>
                    <input
                      id="formViewStoreComposing.formViewComposing.jkl"
                      name="formViewStoreComposing.formViewComposing.jkl"
                      value=""
                    />
                  </div>
                  <br />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
</DocumentFragment>
`;
